### PR TITLE
Restore `simd_shl` and `simd_shr` intrinsics

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

* Restores support for the `simd_shl` and `simd_shr` intrinsics.
* Adds a bunch of tests for them.
* Documents missing overflow check which isn't easy to reproduce.

### Resolved issues:

Towards #1148 
Opens #1963 

### Related RFC:

Optional #ISSUE-NUMBER.

### Testing:

* How is this change tested? Existing regression + ~6 new tests.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/expected/simd-shl-operand-negative/main.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#[repr(simd)]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct i32x2(i32, i32);
+
+#[repr(simd)]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct u32x2(u32, u32);
```
Comment (zhassan-aws) [inline tests/expected/simd-shl-operand-negative/main.rs:17]: This is not used.

File: tests/expected/simd-shl-shift-negative/main.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+#[repr(simd)]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct i32x2(i32, i32);
+
+#[repr(simd)]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct u32x2(u32, u32);
```
Comment (zhassan-aws) [inline tests/expected/simd-shl-shift-negative/main.rs:15]: Same for this.

File: tests/expected/simd-shl-shift-too-large/main.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+#[repr(simd)]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct i32x2(i32, i32);
+
+#[repr(simd)]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct u32x2(u32, u32);
```
Comment (zhassan-aws) [inline tests/expected/simd-shl-shift-too-large/main.rs:15]: Ditto

File: tests/kani/Intrinsics/SIMD/Operators/bitshift.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+#[kani::proof]
+fn test_simd_shl() {
+    let value = kani::any();
+    kani::assume(value >= 0);
+    let values = i32x2(value, value);
+    let shift = kani::any();
+    kani::assume(shift >= 0);
+    kani::assume(shift < 32);
+    let shifts = i32x2(shift, shift);
+    let _result = unsafe { simd_shl(values, shifts) };
```
Comment (zhassan-aws) [inline tests/kani/Intrinsics/SIMD/Operators/bitshift.rs:31]: Can we check the result against that with the normal shift operator `<<`?

File: tests/kani/Intrinsics/SIMD/Operators/bitshift.rs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    kani::assume(shift >= 0);
+    kani::assume(shift < 32);
+    let shifts = i32x2(shift, shift);
+    let _result = unsafe { simd_shr(values, shifts) };
+}
+
+#[kani::proof]
+fn test_simd_shr_unsigned() {
+    let value = kani::any();
+    kani::assume(value >= 0);
```
Comment (zhassan-aws) [inline tests/kani/Intrinsics/SIMD/Operators/bitshift.rs:48]: Is this assumption needed?

File: tests/expected/simd-shl-operand-negative/main.rs (hunk 1 above)
Comment (adpaco-aws) [reply tests/expected/simd-shl-operand-negative/main.rs]: Removed.

File: tests/expected/simd-shl-shift-negative/main.rs (hunk 2 above)
Comment (adpaco-aws) [reply tests/expected/simd-shl-shift-negative/main.rs]: Removed.

File: tests/expected/simd-shl-shift-too-large/main.rs (hunk 3 above)
Comment (adpaco-aws) [reply tests/expected/simd-shl-shift-too-large/main.rs]: Removed.

File: tests/kani/Intrinsics/SIMD/Operators/bitshift.rs (hunk 4 above)
Comment (adpaco-aws) [reply tests/kani/Intrinsics/SIMD/Operators/bitshift.rs]: Yes, it's being done now.

File: tests/kani/Intrinsics/SIMD/Operators/bitshift.rs (hunk 5 above)
Comment (adpaco-aws) [reply tests/kani/Intrinsics/SIMD/Operators/bitshift.rs]: It wasn't, and there was another unneeded assumption in this test. Removed them.

**Review by zhassan-aws — APPROVED** [review]

**Pull request merged**